### PR TITLE
Convert tidier.js to TypeScript && test coverage of tidier.ts

### LINKED ISSUE #3815: Convert tidier.js to TypeScript
Note: Preserved for CodeDay contributors. 

This issue is part of the ongoing work to upgrade CodeMirror from v5 to v6 (see the parent issue for additional context). Alongside this upgrade, the project is also continuing a broader ongoing [TypeScript Migration](https://github.com/processing/p5.js-web-editor/blob/develop/contributor_docs/typescript_migration.md) from JavaScript. 

## Increasing Access
As we're preparing to release the new CodeMirror v6 feature, we'd like to maintain consistency and long-term maintainability to convert files into TypeScript. Breaking down conversion process to happen incrementally through smaller, scoped modules and components can help contributors participate in the migration without needing to immediately understand the entire codebase at once. 

## Feature Details
 The files within the upcoming CM v6 upgrade will need to be converted from JavaScript to TypeScript files. 

This issue specifically focuses on [`tidier.js` file](https://github.com/processing/p5.js-web-editor/blob/develop-codemirror-v6/client/modules/IDE/components/Editor/tidier.js). This file runs `Prettier` on the active `CodeMirror` editor instance by formatting the current file content using `prettier.formatWithCursor` and then updating the editor document while keeping the cursor's position. 

## Suggestions to Start
- Rename `tidier.js` to `tidier.ts`. 
- Add in TypeScript types for the CodeMirror view and the `mode` argument, as well as any other values that will be flagged during the conversion. 
- Update any needed imports/exports. 
- Manually check that "Tidy" and "Format" works for JavaScript, HTML, and CSS files. When formatting the editor content, the cursor should remain stable and not jump to a different location unexpectedly.

### PR DESCRIPTION
**Pull request by kashish00208**
Fixes  #3815 #3816 

Changes:
This PR implements the following changes based on issue #3816  #3815:
1. Rename and TypeScript conversion:
    1. Renamed tidier.js → tidier.ts.
    2. Added TypeScript types for the CodeMirror view, the mode argument, and other values flagged during conversion.
    3. Updated necessary imports/exports.
3. Functionality verification:
    1. Manually verified that Tidy and Format works correctly for JavaScript, HTML, and CSS files.
    2. Ensured that when formatting the editor content, the cursor remains stable and does not jump unexpectedly.
3. Integration tests:
    1. Added tests for tidier.ts 
    2.  Using “Tidy Code” updates the editor document with Prettier formatting.
    3. Cursor position remains stable and predictable after formatting.
    4. Formatting works for HTML, JavaScript, and CSS files.
    5. Tests focus only on integration behavior, not Prettier internals.

I have verified that this pull request:

* [x] has no linting errors (`npm run lint`)
* [x] has no test errors (`npm run test`)
* [x] has no typecheck errors (`npm run typecheck`)
* [x] is from a uniquely-named feature branch and is up to date with the  `develop` branch.
* [x] is descriptively named and links to an issue number, i.e. `Fixes #123`
* [x] meets the standards outlined in the [accessibility guidelines](https://github.com/processing/p5.js-web-editor/blob/develop/contributor_docs/accessibility.md)

### CODE REVIEW

**Review by raclim — CHANGES_REQUESTED** [review]
Thanks so much for your work! This looks great so far! 

We recently merged in a PR that added tests for `tidier.js` as a JavaScript file—would you be open to updating this PR with the changes from the `develop-codemirror-v6` branch, convert that test to TypeScript, and also add in any additional updates to the test that you think would be beneficial?

**Comment by kashish00208** [comment]
> Thanks so much for your work! This looks great so far!
> 
> We recently merged in a PR that added tests for `tidier.js` as a JavaScript file—would you be open to updating this PR with the changes from the `develop-codemirror-v6` branch, convert that test to TypeScript, and also add in any additional updates to the test that you think would be beneficial?

I've updated the PR! I merged the latest changes from develop-codemirror-v6, deleted the JS test file, and fully migrated the suite to TypeScript. I also added the beneficial tests for empty documents and cursor stability do let me know if any chnages needed .

**Comment by raclim** [comment]
Thanks so much for the updates! I think these feel overall good to me—also tagging @khanniie for review!

File: client/modules/IDE/components/Editor/tidier.ts [hunk 1]
```diff
@@ -2,8 +2,17 @@ import prettier from 'prettier/standalone';
 import babelParser from 'prettier/parser-babel';
 import htmlParser from 'prettier/parser-html';
 import cssParser from 'prettier/parser-postcss';
+import type { EditorView } from '@codemirror/view';
+import type { Plugin } from 'prettier';
 
-function prettierFormatWithCursor(parser, plugins, cmView) {
+type parserTypes = 'babel' | 'html' | 'css';
```
Comment (khanniie) [inline client/modules/IDE/components/Editor/tidier.ts]: Small nit but I'm not sure I understand why one type is capitalized and the other isn't! For consistency with the style guide maybe we can call this one ParserTypes?

File: client/modules/IDE/components/Editor/tidier.test.ts [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+  it('formats JavaScript correctly and keeps cursor stable', () => {
+    const messyJs = `function foo(){console.log("hi")}`;
+    const cmView = createEditor(messyJs);
+
+    cmView.dispatch({ selection: { anchor: 14 } });
+
+    tidyCodeWithPrettier(cmView, 'javascript');
+
+    const formattedJs = getDocText(cmView);
+    expect(formattedJs).toContain('console.log("hi");');
```
Comment (khanniie) [inline client/modules/IDE/components/Editor/tidier.test.ts]: for this test, can we can test for the fully properly formatted JavaScript? I don't think this exactly tests for the correctness of the formatter. You should be able to include new lines (\n) in the test string

File: client/modules/IDE/components/Editor/tidier.test.ts [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+  it('formats HTML correctly and keeps cursor stable', () => {
+    const messyHtml = `<div><p>hello</p></div>`;
+    const cmView = createEditor(messyHtml);
+
+    cmView.dispatch({ selection: { anchor: 5 } });
+
+    tidyCodeWithPrettier(cmView, 'html');
+
+    const formattedHtml = getDocText(cmView);
+    expect(formattedHtml).toContain('<p>hello</p>');
```
Comment (khanniie) [inline client/modules/IDE/components/Editor/tidier.test.ts]: Same here can we test for the fully properly formatted output?

File: client/modules/IDE/components/Editor/tidier.test.ts [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+  it('formats CSS correctly and keeps cursor stable', () => {
+    const messyCss = `body{margin:0;padding:0;}`;
+    const cmView = createEditor(messyCss);
+
+    cmView.dispatch({ selection: { anchor: 6 } });
+
+    tidyCodeWithPrettier(cmView, 'css');
+
+    const formattedCss = getDocText(cmView);
+    expect(formattedCss).toContain('margin: 0;');
```
Comment (khanniie) [inline client/modules/IDE/components/Editor/tidier.test.ts]: Here too

**Review by khanniie — CHANGES_REQUESTED** [review]
Overall looking super solid! Thank you so much, this is great! Just a few tiny comments

**Comment by kashish00208** [comment]
@khanniie thank you pointing it out I've updated the tests to be more rigorous. Instead of using .toContain(), I switched to .toBe() to verify the exact output from Prettier .

**Review by khanniie — APPROVED** [review]
LGTM! thank you so much for the changes!
